index.html

```diff
@@ -630,7 +630,51 @@ <h2>DID Documents</h2>
     </section>
 
     <section>
-      <h2>DID Resolvers and DID Resolution</h2>
+      <h2>DID Resolution and DID URL Dereferencing</h2>
+      
+      <p>
+<a>DID resolution</a> is the function of processing a <a>DID</a> using a
+<a>DID method</a> to return a <a>DID document</a> in a compliant representation
+(and additional metadata). 
+A <a>DID resolver</a> is a software and/or hardware component that is capable of
+executing the <a>DID resolution</a> function for at least one
+<a>DID method</a>.
+      </p>
+      
+      <p>
+<a>DID URL dereferencing</a> is the function of processing a <a>DID URL</a> and returning
+a resource, which may or may not be a <a>DID document</a>, along with additional
+metadata. A <a>DID URL dereferencer</a> is a software and/or hardware component that
+is capable of executing the <a>DID URL dereferencing</a> function. Generally, a 
+<a>DID URL dereferencer</a> executes <a>DID resolution</a> on the <a>DID</a> portion of a
+<a>DID URL</a> to obtain a <a>DID document</a>. The <a>DID URL dereferencer</a> then can perform
+additional dereferencing on the <a>DID document</a> using the full <a>DID URL</a>, 
+if necessary.
+      </p>
+      
+      <p>
+A <a>client</a> can call the <a>DID resolution</a> function or the <a>DID URL dereferencing</a>
+function in order to process a <a>DID</a> or <a>DID URL</a>, respectively.
+      </p>
+      
+      <dl>
+        <dt><dfn>client</dfn></dt>
+        <dd>
+Software and/or hardware that invokes a <a>DID resolver</a> in order to execute the <a>DID resolution</a> function and/or a <a>DID URL derfererencer</a> to execute the <a>DID URL dereferencing</a> function. 
+This invocation is done via a <a>binding</a>. The term <a>client</a> does not imply any specific network topology.
+        </dd>
+        <dt><dfn>binding</dfn></dt>
+        <dd>
+A concrete mechanism through which a <a>client</a> invokes a <a>DID resolver</a> or a <a>DID URL dereferencer</a>.
+This could be a local command line tool, a library API, or a network call such as an HTTP(S) request.
+        </dd>
+      </dl>
+      
+      <p>
+The general contracts of the <a>DID resolution</a> and <a>DID URL dereferencing</a> functions
```

Comment:
Using the word "contract" here (and through the rest of the PR) could be confusing. We should use simpler language, like "the inputs to and outputs from the DID resolution and DID URL derefrencing functions are..."

Reply:
I'm fine with that adjustment, I used "contract" here and throughout as it's a well established programming term and was the term by which this WG had been discussing the issue.

Reply:
I've removed the term contract from the pull request.